src/lib/mongoBackend/connectionOperations.cpp

```diff
@@ -0,0 +1,710 @@
+  }
+  catch (const DBException &e)
+  {
+    // FIXME: include wc.nodes() in the output message, + operator doesn't work with integers
+    std::string msg = std::string("setWritteConcern(): ") + /*wc.nodes() +*/
+      " - exception: " + e.what();
+    *err = "Database Error (" + msg + ")";
+    LM_E((err->c_str()));
+    return false;
+  }
+  catch (...)
+  {
+    // FIXME: include wc.nodes() in the output message, + operator doesn't work with integers
+    std::string msg = std::string("setWritteConcern(): ") + /*wc.nodes() + */
+      " - exception: generic";
+    *err = "Database Error (" + msg + ")";
+    LM_E((err->c_str()));
+    return false;
+  }
+
+  return true;
+}
+
+/* ****************************************************************************
+*
+* getWriteConcern -
+*
+*/
+bool getWriteConcern
+(
+  DBClientBase*                  connection,
+  WriteConcern*                  wc,
+  std::string*                   err
+)
+{
+  LM_T(LmtMongo, ("getWriteConcern()"));
+
+  try
+  {
+    *wc = connection->getWriteConcern();
+    LM_I(("Database Operation Successful (getWriteConcern)"));
+  }
+  catch (const DBException &e)
+  {
+    std::string msg = std::string("getWritteConern()") +
+      " - exception: " + e.what();
+    *err = "Database Error (" + msg + ")";
+    LM_E((err->c_str()));
+    return false;
+  }
+  catch (...)
+  {
+    std::string msg = std::string("getWritteConern()") +
+      " - exception: generic";
+    *err = "Database Error (" + msg + ")";
+    LM_E((err->c_str()));
+    return false;
+  }
+
+  return true;
+}
+
+/* ****************************************************************************
+*
+* connectionAuth -
+*
+*/
+extern bool connectionAuth
+(
+  DBClientBase*                  connection,
+  const std::string              db,
+  const std::string              user,
+  const std::string              password,
+  std::string*                   err
+)
+{
+  try {
+    std::string authErr;
+    if (!connection->auth(db, user, password, authErr))
+    {
+      std::string msg = std::string("authentication fails: db=") + db +
+          ", username='" + user + "'" +
+          ", password='*****'" +
+          ", auth_error='" + authErr + "'";
+      *err = "Database Startup Error (" + msg + ")";
+      LM_E((err->c_str()));
+      return false;
+    }
+  }
+  catch (const DBException &e)
+  {
+    std::string msg = std::string("authentication fails: db=") + db +
+        ", username='" + user + "'" +
+        ", password='*****'" +
+        ", expection='" + e.what() + "'";
+    *err = "Database Startup Error (" + msg + ")";
+    LM_E((err->c_str()));
+    return false;
+  }
+  catch (...)
+  {
+    std::string msg = std::string("authentication fails: db=") + db +
+        ", username='" + user + "'" +
+        ", password='*****'" +
+        ", expection=generic";
+    *err = "Database Startup Error (" + msg + ")";
+    LM_E((err->c_str()));
+    return false;
+  }
+
+  return true;
+}
```

Comment:
No, because the req level semaphore is independent of the connectionOperations functions. Eg. some request (such as update context) involved several calls to connectionOperation functions.